mod-source-record-manager-server/src/main/java/org/folio/dao/JobExecutionDaoImpl.java

```diff
@@ -451,4 +459,62 @@ private String buildOrderByClause(List<SortField> sortFields) {
       .collect(Collectors.joining(", ", "ORDER BY ", EMPTY));
   }
 
+  @Override
+  public boolean permanentDeleteJobExecutions(String tenantName) {
+    Promise<RowSet<Row>> promise = Promise.promise();
+    Promise<SQLConnection> connection = Promise.promise();
+    String rollbackMessage = "Rollback transaction. Error during jobExecution deletion process.";
+    Future.succeededFuture()
+      .compose(c -> {
+        pgClientFactory.createInstance(tenantName).startTx(connection);
+        return connection.future();
+      })
+      .compose(s -> fetchJobExecutionIdsToBeConsideredForDeleting(tenantName, connection))
+      .compose(rowSet -> {
+        if (rowSet.rowCount() < 1) {
+          throw new NotFoundException(rollbackMessage);
+        }
+        return mapRowsetValuesToListOfString(rowSet);
+      })
+      .compose(strings -> {
+        Future<RowSet<Row>> jobExecutionProgressFuture = Future.future(rowSetPromise -> deleteJobExecutionsFromParticularTable(tenantName, DELETE_QUERY, connection, PROGRESS_TABLE_NAME, strings.stream().collect(Collectors.joining("','")), JOB_EXECUTION_ID, rowSetPromise));
+        Future<RowSet<Row>> jobMonitoringFuture = Future.future(rowSetPromise -> deleteJobExecutionsFromParticularTable(tenantName, DELETE_QUERY, connection, JOB_MONITORING, strings.stream().collect(Collectors.joining("','")), JOB_EXECUTION_ID, rowSetPromise));
+        Future<RowSet<Row>> jobExecutionSourceChunksFuture = Future.future(rowSetPromise -> deleteJobExecutionsFromParticularTable(tenantName, DELETE_QUERY, connection, JOB_EXECUTION_SOURCE_CHUNKS, strings.stream().collect(Collectors.joining("','")), JOBEXECUTIONID, rowSetPromise));
+        Future<RowSet<Row>> journalRecordsFuture = Future.future(rowSetPromise -> deleteJobExecutionsFromParticularTable(tenantName, DELETE_QUERY, connection, JOURNAL_RECORDS, strings.stream().collect(Collectors.joining("','")), JOB_EXECUTION_ID, rowSetPromise));
+        Future<RowSet<Row>> jobExecutionFuture = Future.future(rowSetPromise -> deleteJobExecutionsFromParticularTable(tenantName, DELETE_QUERY, connection, TABLE_NAME, strings.stream().collect(Collectors.joining("','")), ID, rowSetPromise));
+        return CompositeFuture.all(jobExecutionProgressFuture, jobMonitoringFuture, jobExecutionSourceChunksFuture, journalRecordsFuture, jobExecutionFuture);
+      })
+      .compose(updateHandler -> {
+        Promise<Void> endTxFuture = Promise.promise();
+        pgClientFactory.createInstance(tenantName).endTx(connection.future(), endTxFuture);
+        return endTxFuture.future();
+      }).onComplete(ar -> {
+        if (ar.failed()) {
+          pgClientFactory.createInstance(tenantName).rollbackTx(connection.future(), rollback -> promise.fail(ar.cause()));
+          return;
+        }
+        promise.complete();
+      });
+    return promise.tryComplete();
```

Comment:
Basically it mismatches java doc contract for this method  in some way, instead of saying that deletion competed with errors or not - it says does deletion process completed or still in progress

Reply:
Hi @SerhiiNosko - Updated code as per your suggestion. Thank you 🙂